libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py

```diff
@@ -55,25 +60,42 @@
 
 
 class TokenExchangeState(Model):
+    """TokenExchangeState
+
+    :param connection_name: The connection name that was used.
+    :type connection_name: str
+    :param conversation: Gets or sets a reference to the conversation.
+    :type conversation: ~botframework.connector.models.ConversationReference
+    :param relates_to: Gets or sets a reference to a related parent conversation for this token exchange.
+    :type relates_to: ~botframework.connector.models.ConversationReference
+    :param bot_ur: The URL of the bot messaging endpoint.
+    :type bot_ur: str
+    :param ms_app_id: The bot's registered application ID.
+    :type ms_app_id: str
+    """
+
     _attribute_map = {
         "connection_name": {"key": "connectionName", "type": "str"},
         "conversation": {"key": "conversation", "type": "ConversationReference"},
-        "bot_url": {"key": "botUrl", "type": "str"},
+        "relates_to": {"key": "relatesTo", "type": "ConversationReference"},
+        "bot_url": {"key": "connectionName", "type": "str"},
         "ms_app_id": {"key": "msAppId", "type": "str"},
     }
 
     def __init__(
         self,
         *,
         connection_name: str = None,
-        conversation: ConversationReference = None,
+        conversation=None,
```

Comment:
other parameters use spaces around the equals, but your new ones here don't. Please add the spaces for consistency.

Reply:
Yes, this is also part of the formatting, the convention is that if you use typing the format is:

`variable: VariableType = value`

but if you don't use typing the format is:

`variable=value`

Reply:
Formatting aside, I removed the typing accidentally... I should put it back